### ISSUE TITLE
Adding padding to ValueAxis

### ISSUE THREAD

**Issue by andrewbridge** [amcharts/amcharts4#1523]
This was previously brought up in an [issue a few months ago](https://github.com/amcharts/amcharts4/issues/1279) but the author never responded to the reply.

While `startLocation` and `endLocation` are available to `CategoryAxis` and `DateAxis` ([as described in the docs](https://www.amcharts.com/docs/v4/concepts/axes/positioning-axis-elements/#Setting_where_axis_starts_and_ends)), it does not seem to be honoured by a `ValueAxis`, nor is it explained why it is not honoured.

The results achieved in one of the [codepen examples](https://codepen.io/team/amcharts/pen/erQovR) is exactly what I'm looking to achieve, however as my data set consists of numbers rather than dates or discrete categories, I'm unable to use either `Axis` type that does consider the `startLocation` and `endLocation`.

I've set up a [reduced test case](https://codepen.io/andrewbridge/pen/ZgJddZ?editors=0010) that produces the chart I'm having problems with. Here the chart appears to be given just above the [default minimum grid distance of 120](https://www.amcharts.com/docs/v4/concepts/axes/positioning-axis-elements/#Setting_the_density_of_the_the_grid_labels) and as the dataset fits exactly between a range of 60, amcharts sets up a grid of 5 and places the chart in the middle.

![image](https://user-images.githubusercontent.com/2941908/62376137-e2897a00-b537-11e9-95ae-841f44e5b69b.png)

This means 2/5 of the chart is wasted however, and I'd like the chart to extend to both sides.

On L11 and L12, I've commented out the two methods I can see of doing this. First, `strictMinMax`:

![image](https://user-images.githubusercontent.com/2941908/62376251-254b5200-b538-11e9-9bda-25936c7cfe9b.png)

But now the bullet points are partially obscured, making it hard to hover over them to trigger the tooltip.

Second, `minGridDistance`:

![image](https://user-images.githubusercontent.com/2941908/62376312-490e9800-b538-11e9-83ee-fcdce7ac82ac.png)

While this provides the perfect amount of space for the bullets to be clear of the axes, it introduces a whole set of new problems to fix, such as label density, grid line density etc.

The fuller example I'm looking to achieve has many more bullets as they're integral to the data we're displaying:

![image](https://user-images.githubusercontent.com/2941908/62376484-a3a7f400-b538-11e9-87aa-a861fca876b9.png)

Is there any alternative to the methods I've found or any chance of `startLocation` and `endLocation` being added for `ValueAxis`? Any help would be hugely appreciated as I've reached a dead end!

**Comment by martynasma** [comment]
For a `ValueAxis` the two properties `extraMin` and `extraMax` are your friends:

https://www.amcharts.com/docs/v4/concepts/axes/value-axis/#Relatively_expanding_the_scale

**Comment by andrewbridge** [comment]
Hi @martynasma, thank for the quick reply, your help's much appreciated!

I actually went straight for `extraMin` and `extraMax` but based on playing about with those attributes, it looks like these attributes are applied after or accounted for during the auto-fitting process or not at all if `strictMinMax` is used, is that right?

[I've forked my demo](https://codepen.io/andrewbridge/pen/gVvjPQ?editors=0010) to use `extraMin` and `extraMax` but it appears to squish the chart further. I believe the default value for both properties is 0, so presumably increasing the value at all will only add space. If I give the properties negative values, it actually obscures some of the values from view and the space previously added during the auto-fitting process is no longer there.

![image (5)](https://user-images.githubusercontent.com/2941908/62605311-bc762800-b8f1-11e9-93a5-de25156e89a8.png)
_Comparison of graph with and without `extraMin` and `extraMax` properties changed_

![image](https://user-images.githubusercontent.com/2941908/62605393-eaf40300-b8f1-11e9-814f-0b26e3512686.png)
_Graph with negative `extraMin` and `extraMax` values_

Is there anything else I can try?

**Comment by martynasma** [comment]
Using negative values for `extraMin`/`extraMax` does not make a lot of sense to me.

Back to your demo, you have very large numbers set (0.5) it increases the scale of the actual value axis scale by 2. Why don't you try something more moderate, like 0.2?

**Comment by andrewbridge** [comment]
No, setting a negative to something named "extra" didn't make sense to me either, but it's interesting to see its effect.

Apologies, I used 0.5 as a drastic example to make the change clearer. I returned to the codepen I posted earlier and changed the values to 0.1:
![image](https://user-images.githubusercontent.com/2941908/62614509-7629c480-b903-11e9-8f30-8942cb10bd27.png)

As you can see, there isn't a lot of change, and if anything, the chart is slightly more squished.

As I say, I believe the default value of both `extraMin` and `extraMax` is 0, so my assumption is that increasing this value will only decrease the space that the line takes up on the chart.

**Comment by martynasma** [comment]
You may need to increase granularity of axis grid/values so it can "cur off" the scale at finer numbers.

Use `minGridDistance`.

https://www.amcharts.com/docs/v4/concepts/axes/positioning-axis-elements/#Setting_the_density_of_the_the_grid_labels

**Comment by andrewbridge** [comment]
You're right, I found this option in the docs too and [tried it in my original demo](https://codepen.io/andrewbridge/pen/ZgJddZ?editors=0010). But to get the granularity I need (i.e, the bullets just inside the chart area), I'd have to decrease the `minGridDistance` to a very small value , which then creates further problems with too many grid lines and too many labels.

![image](https://user-images.githubusercontent.com/2941908/62376312-490e9800-b538-11e9-83ee-fcdce7ac82ac.png)

I guess there are ways to fix these problems but it seems like I'm fixing a new problem I've created needlessly.

It's frustrating that the `startLocation` and `endLocation` properties aren't available just because of the data type in my dataset. The [demo in the docs for these properties](https://codepen.io/team/amcharts/pen/erQovR) is _exactly_ what I'm trying to achieve so the bullets at either end of this chart aren't partially obscured.

Is there any other alternative or are there any plans to add support for these properties to the `ValueAxis`?
